files/justfiles/hardening.just

```diff
@@ -14,6 +14,88 @@ harden-flatpak FLATPAK="":
+    echo "This will configure flatpak to automatically reject most permissions (with the exception of the Wayland socket and the Dri device, since these are commonly used and ensure at the very least most apps will work without crashing)."
+    echo "This will also grant Flatseal access to certain permissions to allow it to operate and make reconfiguring much easier."
+    echo "NOTE: This will break just about all Flatpaks by default, it is ON YOU to configure them to work with this configuration."
+    echo "NOTE 2: This DOES NOT enable hardened_malloc, use the harden-flatpak ujust command."
+
+    echo ""
+    echo "-- Share Permissions --"
+    for i in ${kSharePermissions[@]}; do
+        echo "	Rejecting $i..."
+        $kCommand --unshare=$i
+    done
+    echo ""
+    echo "-- Socket Permissions --"
+    for i in ${kSocketPermissions[@]}; do
+        echo "	Rejecting $i..."
+        $kCommand --nosocket=$i
+    done
+    echo ""
+    echo "-- Device Permissions --"
+    for i in ${kDevicePermissions[@]}; do
+        echo "	Rejecting $i..."
+        $kCommand --nodevice=$i
+    done
+    echo ""
+    echo "-- Feature Permissions --"
+    for i in ${kFeaturePermissions[@]}; do
+        echo "	Rejecting $i..."
+        $kCommand --disallow=$i
+    done
+    echo ""
+    echo "-- Filesystem Permissions --"
+    for i in ${kFilesystemPermissions[@]}; do
+        echo "	Rejecting $i..."
+        $kCommand --nofilesystem=$i
+    done
+    echo ""
+    echo "-- Dangerous Filesystem Permissions --"
+    echo "Note: This is a VERY flawed implementation but it does cover a few blatant sandbox escape methods (such as the .bashrc escape or mounted drive access)"
+    echo "It is not possible to cover all files since each file can be requested manually and therefore must be rejected manually"
+    for i in ${kDangerousFilesystemPermissions[@]}; do
+        echo "	Rejecting $i..."
+        $kCommand --nofilesystem=$i
+    done
+    echo ""
+    echo "NOTE: The next 2 lockdowns are NOT complete and only cover a list of known names, this can be expanded at any time"
+    echo "-- Session Bus Name Access --"
+    for i in ${kKnownSessionBusNames[@]}; do
+        echo "	Rejecting $i..."
+        $kCommand --no-talk-name=$i
+
+    echo "-- System Bus Name Access --"
+    for i in ${kKnownSystemBusNames[@]}; do
+        echo "	Rejecting $i..."
+        $kCommand --system-no-talk-name=$i
+    done
+    echo ""
+    echo "-- Peristent Filesystem Grant --"
+    echo "Note: This is to unbreak many Flatpaks by allowing the app to store persistent data in their own, isolated home directory without accessing the user's"
+    echo "	Granting access to peristent home..."
+    $kCommand --perist=.
+    echo ""
+    echo "-- Granting Flatseal Access to Bus Names --"
+    for i in ${kFlatsealNameAccess[@]}; do
+        echo "	Rejecting $i..."
+        $kCommand --talk-name=$i com.github.tchx84.Flatseal
+    done
+    echo ""
+    echo "Done"
+
```

Comment:
`--device=dri` can and should be added by default, some apps rely on getting it via `device=all` and break without it. Practically all apps have and need this permission in some form.